From: Ben Hutchings <ben@decadent.org.uk>
Subject: perf: Avoid ABI change in 3.9.8
Date: Sat, 29 Jun 2013 17:57:25 +0100
Forwarded: not-needed

mmap-related fields in struct perf_event moved to struct ring_buffer.
struct perf_event is exposed to modules, but they probably didn't
use those fields, so just add them back.  struct ring_buffer is not
exposed to modules, so just hide that change from genksyms.



--- a/include/linux/perf_event.h
+++ b/include/linux/perf_event.h
@@ -405,6 +405,10 @@ struct perf_event {
 	struct mutex			mmap_mutex;
 	atomic_t			mmap_count;
 
+	/* unused, kept for ABI compatibility */
+	int				mmap_locked;
+	struct user_struct		*mmap_user;
+
 	struct ring_buffer		*rb;
 	struct list_head		rb_entry;
 
--- a/kernel/events/internal.h
+++ b/kernel/events/internal.h
@@ -31,9 +31,11 @@ struct ring_buffer {
 	spinlock_t			event_lock;
 	struct list_head		event_list;
 
+#ifndef __GENKSYMS__
 	atomic_t			mmap_count;
 	unsigned long			mmap_locked;
 	struct user_struct		*mmap_user;
+#endif
 
 	struct perf_event_mmap_page	*user_page;
 	void				*data_pages[0];
